### PR TITLE
Adding termination signal to be run on the container through 'docker …

### PR DESCRIPTION
**Pull request by derosm2**
…exec'.  See issue #18

### CODE REVIEW

**Comment by derosm2** [comment]
#18

File: health/termination.py [hunk 1]
```diff
@@ -31,3 +31,23 @@ def run(self, container, report):
...
     return True
+
+class ExecTerminationSignal(TerminationSignal):
+  """ A termination signal which tries to EXEC a command on a running container """
+  def __init__(self, config):
+    super(TerminationSignal, self).__init__()
+    self.config = config
+
+  def run(self, container, report):
+    report('ExecTerminationSignal in container ' + container['Id'][0:12] ,
```
Comment (josephschorr) [inline health/termination.py]: Use Python string replacement here:

```
'Foo %s %s' % (first_value, second_value)
```

File: health/termination.py [hunk 2]
```diff
@@ -31,3 +31,23 @@ def run(self, container, report):
...
+           level=ReportLevels.EXTRA)
+    report('Exec: ' + self.config.command)
```
Comment (josephschorr) [inline health/termination.py]: Ditto

Comment (josephschorr) [inline health/termination.py]: Newline before this call

File: health/checks.py [hunk 3]
```diff
@@ -23,6 +23,7 @@ def buildHealthCheck(check_config):
 TERMINATION_SIGNALS = {
   'http': partial(HttpTerminationSignal, 'http'),
   'https': partial(HttpTerminationSignal, 'https'),
+  'exec': ExecTerminationSignal
```
Comment (josephschorr) [inline health/checks.py]: Comma at the end of this line

File: config/GantryConfig.py [hunk 4]
```diff
@@ -28,6 +28,7 @@ class _TerminationSignal(CFObject):
   id = CFField('id').default('').name_field()
   kind = CFField('kind').value_field()
   timeout = CFField('timeout').kind(int).default(3)
+  command = CFField('command').default('')
```
Comment (josephschorr) [inline config/GantryConfig.py]: Call this `exec_command`

**Comment by josephschorr** [comment]
So to understand at a meta level: With this change, you can specify a command to be run against the container before it shuts down, to ensure it can clean itself up?

**Comment by derosm2** [comment]
That's right. Thanks for the notes, should have the changes up soon.

File: health/termination.py (hunk 2 above)
Comment (derosm2) [inline health/termination.py]: Ended up just combining the two log statements to simplify things.

**Comment by josephschorr** [comment]
LGTM

**Pull request merged**